query/src/main/java/org/infinispan/query/affinity/ShardAllocationManagerImpl.java

```diff
@@ -69,7 +69,7 @@ public String getShardFromKey(Object key) {
    @Override
    public void initialize(Integer numberOfShards, int numSegments) {
       this.numSegments = numSegments;
-      this.numShards = numberOfShards == null ? numSegments : numberOfShards;
+      this.numShards = numberOfShards;
```

Comment:
numberOfShards is Integer, not int, so it can still be null in theory. Maybe it's worth keeping the old line here or change numberOfShards from Integer to int.